packages/explorer/src/observer/decorator.ts

```diff
@@ -22,49 +31,225 @@ export function observer({ explorerUrl = "http://localhost:13690", waitForTransa
 ) => Pick<WalletActions<chain, account>, "writeContract"> {
   const emit = createBridge({ url: `${explorerUrl}/internal/observer-relay` });
 
-  return (client) => ({
-    async writeContract(args) {
-      const writeId = `${client.uid}-${++writeCounter}`;
-      const write = getAction(client, writeContract, "writeContract")(args);
-
-      // `writeContract` above will throw if this isn't present
-      const functionAbiItem = getAbiItem({
-        abi: args.abi,
-        name: args.functionName,
-        args: args.args,
-      } as never)!;
-
-      emit("write", {
-        writeId,
-        address: args.address,
-        from: client.account!.address,
-        functionSignature: formatAbiItem(functionAbiItem),
-        args: (args.args ?? []) as never,
-        value: args.value,
-      });
-      Promise.allSettled([write]).then(([result]) => {
-        emit("write:result", { ...result, writeId });
-      });
-
-      write.then((hash) => {
-        const receipt = getAction(client, waitForTransactionReceipt, "waitForTransactionReceipt")({ hash });
-        emit("waitForTransactionReceipt", { writeId, hash });
-        Promise.allSettled([receipt]).then(([result]) => {
-          emit("waitForTransactionReceipt:result", { ...result, writeId });
+  return (client) => {
+    console.log("observerWrite client:", client);
+
+    // if (args.address.toLowerCase() === entryPoint07Address.toLowerCase()) {
+    //   console.log("observerWrite handleOps", args);
+    //   console.log("observerWrite args.args", args.args[0]);
+
+    //   const arg = args.args[0][0];
+    //   try {
+    //     console.log("observerWrite arg", arg.callData);
+
+    //     const decoded = decodeFunctionData({
+    //       abi: parseAbi(["function execute(address target, uint256 value, bytes calldata data)"]),
+    //       data: arg.callData,
+    //     });
+
+    //     console.log("observerWrite decoded", decoded);
+
+    //     const decodedWorldCall = decodeFunctionData({
+    //       abi: IBaseWorldAbi,
+    //       data: decoded.args[2],
+    //     });
+
+    //     console.log("observerWrite decodedWorldCall", decodedWorldCall);
+    //   } catch (error) {
+    //     console.error("observerWrite error", error);
+    //   }
+    // }
+
+    if (client.type === "bundlerClient") {
```

Comment:
shouldn't need to and prob shouldn't conditionally extend the client based on the type

Reply:
`sendUserOperation` cannot be extended on a wallet client, hence why. But this might still be worked around differently